docs/rfcs/001-recent-commits.md

```diff
@@ -0,0 +1,87 @@
+* A "Soft reset" option. Choosing this performs a `git reset --soft` which moves `HEAD` to the chosen commit and populates the staged changes list with all of the cumulative changes from all commits between the chosen one and the previous `HEAD`.
+
+On click, reveal a tool-tip containing:
+
+* Additional user information consistently with the GitHub integration's user mention item.
+* The full commit message and body.
+* The absolute timestamp of the commit.
+* Navigation button ("open" to a git show-ish pane item)
+* Action buttons ("amend" on the most recent commit, "revert", and "reset" with "hard", "mixed", and "soft" suboptions)
+
+![commit-popout](https://user-images.githubusercontent.com/17565/36570682-11545cae-17e8-11e8-80a8-ffcf7328e214.JPG)
+
+If the Git dock item is dragged to the bottom dock, the recent commit section will remain a vertical list but appear just to the right of the mini commit editor.
+
+![bottom-dock](https://user-images.githubusercontent.com/17565/36570687-14738ca2-17e8-11e8-91f7-5cf1472d871b.JPG)
+
+## Drawbacks
+
+Consumes vertical real estate in Git panel.
+
+The "undo" button is not a native git concept. This can be mitigated by adding a tooltip to the "undo" button that defines its action: a `git reset` and commit message edit.
+
+The "soft reset" and "hard reset" context menu options are useful for expert git users, but likely to be confusing. It would be beneficial to provide additional information about the actions that both will take.
+
+The modal dialog on "hard reset" is disruptive considering that the lost changes are recoverable from `git reflog`. We may wish to remove it once we visit a reflog view within the package. Optionally add "Don't show" checkbox to disable modal.
+
+## Rationale and alternatives
+
+- Display tracking branch in separator that indicates which commits have been pushed. This could make the purpose of the divider more clear. Drawback is that this takes up space.
+
+## Unresolved questions
+
+- Allow users to view the changes introduced by recent commits. For example, interacting with one of the recent commits could launch a pane item that showed the full commit body and diff, with additional controls for reverting, discarding, and commit-anchored interactions.
+- Providing a bridge to navigate to an expanded log view that allows more flexible and powerful history exploration.
+- Show an info icon and provide introductory information when no commits exist yet.
+- Add a "view diff from this commit" option to the recent commit context menu.
+- Integration with and navigation to "git log" or "git show" pane items when they exist.
```

Comment:
How do we surface that we've made a commit for you before doing a `git reset --hard`? So you never actually lose any unstaged changes

Reply:
Right-click context menu on history? "Undo reset" until next commit is made.